mythril/analysis/security.py

```diff
@@ -45,42 +51,55 @@ def get_detection_module_hooks(modules, hook_type="pre"):
     return dict(hook_dict)
 
 
-def get_detection_modules(entrypoint, include_modules=()):
+def get_detection_modules(entrypoint, include_modules=(), custom_modules_directory=""):
     """
 
     :param entrypoint:
     :param include_modules:
     :return:
     """
-
     module = importlib.import_module("mythril.analysis.modules.base")
     module.log.setLevel(log.level)
 
     include_modules = list(include_modules)
 
     _modules = []
 
-    if not include_modules:
-        for loader, module_name, _ in pkgutil.walk_packages(modules.__path__):
-            if module_name != "base":
+    custom_modules_path = os.path.abspath(custom_modules_directory)
+
+    if custom_modules_directory and custom_modules_path not in sys.path:
+        sys.path.append(custom_modules_path)
+
+    custom_packages = (
+        list(pkgutil.walk_packages([custom_modules_path]))
+        if custom_modules_path
+        else []
+    )
+    packages = list(pkgutil.walk_packages(modules.__path__)) + custom_packages
+
+    for loader, module_name, _ in packages:
+        if include_modules and module_name not in include_modules:
+            continue
+
+        if module_name != "base":
+            try:
                 module = importlib.import_module(
                     "mythril.analysis.modules." + module_name
                 )
-                module.log.setLevel(log.level)
-                if module.detector.entrypoint == entrypoint:
-                    _modules.append(module)
-    else:
-        for module_name in include_modules:
-            module = importlib.import_module("mythril.analysis.modules." + module_name)
-            if module.__name__ != "base" and module.detector.entrypoint == entrypoint:
-                module.log.setLevel(log.level)
+            except ModuleNotFoundError:
+                try:
+                    module = importlib.import_module(module_name, custom_modules_path)
```

Comment:
Hmm... I hadn't noticed that it's a different syntax to import a module from a custom path.

I'm wondering about the situation where a custom module has the same name as a non-custom module. Ehh, we can probably just pretend that it will never happen.